utils/prisma/prismaHelpers.ts

```diff
@@ -0,0 +1,47 @@
+import { Prisma } from '@prisma/client';
+import { NextApiResponse } from 'next';
+import { StatusMessageType } from '../../types/apiResponse';
+import { createJsonResponse, HTTP_STATUS_BAD_REQUEST, HTTP_STATUS_INTERNAL_SERVER_ERROR } from '../http/httpHelper';
+
+/**
+ * Utility function that takes care of Prisma runtime errors during CREATE.
+ *
+ * @param res The NextApiResponse object.
+ * @param creator The Prisma Creator function.
+ */
+export function createIfPossible(res: NextApiResponse, creator: () => Promise<void>) {
+  creator().catch((error) => {
+    console.error(error);
+
+    if (error instanceof Prisma.PrismaClientKnownRequestError) {
+      res.status(HTTP_STATUS_BAD_REQUEST).json(
+        createJsonResponse({}, [
+          {
+            type: StatusMessageType.Error,
+            message: 'A database error has occured due to the request. Please try again.',
```

Comment:
```suggestion
            message: 'A database error has occured due to the request.',
```
Would asking them to try again help? Since it's a database error (which probably means there is some validation error or database constraint violation)  

Not sure if we can add more information to this error message (in future perhaps) since it implies that there is something the user can probably do to fix?

Reply:
Not necessarily, could just be a db conn limit, in which case trying again (after a while) will work.